### PR TITLE
Aqaint 11

### PR DESCRIPTION
**Pull request by EKustova**

### CODE REVIEW

File: ChainReactionBindings/Bindings/SearchPageBindings.cs [hunk 1]
```diff
@@ -116,5 +114,60 @@ public void ClickBuyButtonOnItem()
...
+		{
+			page.FromElement.SendKeys(from);
+			page.ToElement.SendKeys(to);
+			page.GoButtonElement.Click();
+		}
+
+		[Then(@"I see message ""(.*)"" after filter on search page")]
+		public void ThenISeeMessageAfterFilterOnSearchPage(string text)
+		{
+			Assert.AreEqual(text, page.NoResultsFilterMessageElement.Text);
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/SearchPageBindings.cs]: Use Assert.That; also please add message

File: ChainReactionBindings/Bindings/SearchPageBindings.cs [hunk 2]
```diff
@@ -48,19 +48,17 @@ public void ThenISeeMessageOnSearchPage(string text, string searchText)
...
 		public void ThenISeeOnSearchPage(string text)
 		{
 			Assert.AreEqual(text, page.CategoryResultElement.Text);
-
 		}
 		[Then(@"I see ""(.*)"" items on search page")]
 		public void ThenISeeItemsOnSearchPage(int count)
 		{
-			Assert.AreEqual(count, page.ItemsCountElement.FindElements(By.ClassName("products_details_container")).Count);
+			Assert.AreEqual(count, page.GetCountItems());
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/SearchPageBindings.cs]: Please use Assert.That

File: ChainReactionBindings/Bindings/SearchPageBindings.cs [hunk 3]
```diff
@@ -116,5 +114,60 @@ public void ClickBuyButtonOnItem()
...
+		[When(@"I click colour ""(.*)"" on search page")]
+		public void ClickColourOnSearchPage(string colour)
+		{
+			page.ColourClick(colour);
+		}
+
+		[Then(@"I see items include colour ""(.*)""")]
+		public void ThenISeeItemsIncludeColour(string colour)
+		{
+			var itemsDesc = page.ItemsCountElement.FindElements(By.ClassName(colour.ToLower()));
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/SearchPageBindings.cs]: Looks like we don't need to create variable here

File: ChainReactionBindings/Bindings/SearchPageBindings.cs [hunk 4]
```diff
@@ -116,5 +114,60 @@ public void ClickBuyButtonOnItem()
...
+		}
+
+		[When(@"I enter price from ""(.*)"" to ""(.*)"" on search page and click Go button")]
+		public void EnterPriceFromToOnSearchRageAndClickGoButton(string from, string to)
+		{
+			page.FromElement.SendKeys(from);
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/SearchPageBindings.cs]: Can we hide these three lines in some nice function inside the page object?

File: ChainReactionBindings/Bindings/TopMenuBindings/TopMenuBindings.cs [hunk 5]
```diff
@@ -64,6 +57,7 @@ public void SeeBasketPopupIsOpen()
 		public void SeeMessageInBasketPopup(string message)
 		{
 			Assert.AreEqual(message, page.BasketMessageElement.Text);
+			Thread.Sleep(TimeSpan.FromSeconds(5));
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/TopMenuBindings/TopMenuBindings.cs]: Consider using explicit feature file step for waiting, it's not good to hide this sleep logic

File: ChainReactionBindings/Bindings/ItemDetailsPageBindings.cs [hunk 6]
```diff
@@ -141,19 +141,13 @@ public void ThenISeeSizeChartIsOpened()
 		[When(@"I scroll to element")]
 		public void ThenIScrollToElement()
 		{
-			var element = page.ContainerElement.FindElement(By.ClassName("crcPDPList"));
-			Actions act = new Actions(Browser.Driver);
-			act.MoveToElement(element);
-			act.Perform();
+			page.ScrollToElement();
```
Comment (etolmachev) [inline ChainReactionBindings/Bindings/ItemDetailsPageBindings.cs]: Rename ScrollToElement func and rename step to make it more readable

**Review by etolmachev — CHANGES_REQUESTED** [review]

File: ChainReactionBindings/Bindings/ItemDetailsPageBindings.cs (hunk 6 above)
Comment (EKustova) [reply ChainReactionBindings/Bindings/ItemDetailsPageBindings.cs]: done

File: ChainReactionBindings/Bindings/SearchPageBindings.cs (hunk 2 above)
Comment (EKustova) [reply ChainReactionBindings/Bindings/SearchPageBindings.cs]: done

File: ChainReactionBindings/Bindings/SearchPageBindings.cs (hunk 4 above)
Comment (EKustova) [reply ChainReactionBindings/Bindings/SearchPageBindings.cs]: done

File: ChainReactionBindings/Bindings/SearchPageBindings.cs (hunk 1 above)
Comment (EKustova) [reply ChainReactionBindings/Bindings/SearchPageBindings.cs]: done

**Pull request merged**